harness/determined/common/experimental/metrics.py

```diff
@@ -8,16 +8,16 @@
 
 @dataclasses.dataclass
 class TrialMetrics:
-    """
-    Specifies a metric that the trial reported.
+    """Specifies a metric that the trial reported.
 
     Attributes:
-        trial_id
-        trial_run_id
-        steps_completed
-        end_time
-        metrics
-        batch_metrics
+        trial_id: The ID of the trial that reported the metric.
+        trial_run_id: The ID of the trial run that reported the metric.
+        steps_completed: The number of steps that the trial had completed when the metric was
+            reported.
+        end_time: The time when the metric was reported.
+        metrics: A dict of metrics that the trial reported.
+        batch_metrics: A dict of metrics that the trial reported for each batch.
```

Comment:
can we also somehow mark `batch_metrics` as deprecated?

Reply:
For sure. Do you know anything about the deprecation story (or happen to have any easy pointers to get me started)? I can comb git logs to try to figure it out if not.

Reply:
I don't see anything in the protos ([TrialMetrics](https://github.com/determined-ai/determined/blob/8d2b5ce5b2134513a3c559d5545ad3e8c4fb3230/proto/src/determined/trial/v1/trial.proto#L240) or [Metrics](https://github.com/determined-ai/determined/blob/8d2b5ce5b2134513a3c559d5545ad3e8c4fb3230/proto/src/determined/common/v1/common.proto#L11)) that imply `batch_metrics` is deprecated. Continuing to look.

Reply:
Nor in the [sql](https://github.com/determined-ai/determined/blob/8d2b5ce5b2134513a3c559d5545ad3e8c4fb3230/master/static/srv/get_checkpoints_for_experiment.sql#L21) or [master](https://github.com/determined-ai/determined/blob/8d2b5ce5b2134513a3c559d5545ad3e8c4fb3230/master/internal/db/postgres_trial_metrics.go#L33).

Reply:
Are we actually deprecating this? I actually just don't know. I remember having to ignore it when I was using this for inference, but I don't know if it's used it other use cases.